Refactor and fix HTMLTrackElement/src.html

zcorpan opened the pull request:
Reflecting USVString attributes containing a URL should return
the empty string if the content attribute's value is empty.
Ref. https://github.com/whatwg/html/issues/859

A single \u0000 character gets stripped by the URL parser;
"Remove any leading and trailing C0 controls and space from input."

Fixes #2125. Closes #2663.

---

cc @tkent-google @bzbarsky

zcorpan committed:
Refactor and fix HTMLTrackElement/src.html

Reflecting USVString attributes containing a URL should return
the empty string if the content attribute's value is empty.
Ref. https://github.com/whatwg/html/issues/859

A single \u0000 character gets stripped by the URL parser;
"Remove any leading and trailing C0 controls and space from input."

Fixes #2125. Closes #2663.

bzbarsky commented:
Do we have any implementor buy-in at all on the empty string behavior the spec added in https://github.com/whatwg/html/issues/859 ?

zcorpan commented:
@tkent-google ^

I'm happy to revert the empty string thing in the spec if there is little interest in changing it. 

Demo http://software.hixie.ch/utilities/js/live-dom-viewer/?saved=4442

zcorpan committed:
Change empty string case to expect resolved URL

Ref.
https://github.com/whatwg/html/pull/1745
https://github.com/whatwg/html/issues/1739

domenic commented:
Looks like at least part of this needs revision before merging.

zcorpan commented:
Which part? Note that I fixed the empty string case to match the current spec in https://github.com/w3c/web-platform-tests/pull/3639/commits/2f5f34515cc9e0685f554236231250c0f8a49595

domenic commented:
Oh, doh, sorry I missed that.

domenic commented:
LGTM, will let you merge with updated commit message
